src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -470,6 +470,27 @@ void Unit::MonsterMoveWithSpeed(float x, float y, float z, float speed, bool gen
     init.Launch();
 }
 
+void Unit::MoveAdvanceTo(Unit* target)
+{
+    if (!target) // Just in case
+        return;
+
+    // Do not reposition ourself when we are not allowed to move
+    if (IsMovementPreventedByCasting() || isMoving() || !CanFreeMove())
+        return;
+
+    float x, y, z;
+    GetNearPoint(target, x, y, z, 0.0f, 0.0f, target->GetAngle(this));
+    Movement::MoveSplineInit init(this);
+    init.MoveTo(x, y, z, true, false);
+
+    // Beasts move backwards instead of turning arround
+    if (ToCreature() && ToCreature()->GetCreatureTemplate()->type == CREATURE_TYPE_BEAST)
```

Comment:
Not strictly limited to beasts, more like beast models. Example: http://www.wowhead.com/npc=91847/risen-hound also moves backwards (undead dog)

Reply:
Any idea how to detect this?

Reply:
not really atm, maybe there is something in creaturemodel/displayinfo related db2s

Reply:
Hmm, I rememember the automatic trigger model detection system is using something like this to determine the triger models. Would that be an acceptable solution?

Reply:
@Shauren are you sure thats also for WotLk? Your URL refers to an ID which indicates it was added after WotLk (exactly at patch 7.0.3.22290). I don't remember any creature which was an exception in this case (in WotLk).